src/participant_network/participant_network.star

```diff
@@ -52,12 +124,72 @@ def launch_participant_network(num_participants, network_params):
 
 	)
 
-	print(json.indent(json.encode(cl_data)))
+	print(json.indent(json.encode(cl_genesis_data)))
+
+	print("Launching CL network")
+
+	cl_launchers = {
+		module_io.CLClientType.lighthouse : {"launcher": new_lighthouse_launcher(cl_genesis_data), "launch_method": launch_lighthouse},
+		module_io.CLClientType.lodestar: {"launcher": new_lodestar_launcher(cl_genesis_data), "launch_method": launch_lodestar},
+		module_io.CLClientType.nimbus: {"launcher": new_nimbus_launcher(cl_genesis_data), "launch_method": launch_nimbus},
+		module_io.CLClientType.prysm: {"launcher": new_prysm_launcher(cl_genesis_data, cl_validator_data.prysm_password_relative_filepath, cl_validator_data.prysm_password_artifact_uuid), "launch_method": launch_prysm},
+		module_io.CLClientType.teku: {"launcher": new_teku_launcher(cl_genesis_data), "launch_method": launch_teku},
+	}
+
+	all_cl_client_contexts = []
+	all_mevboost_contexts = []
+	preregistered_validator_keys_for_nodes = cl_validator_data.per_node_keystores
+
+	for index, participant in enumerate(participants):
+		cl_client_type = participant.cl_client_type
+
+		if cl_client_type not in cl_launchers:
+			fail("Unsupported launcher '{0}', need one of '{1}'".format(cl_client_type, ",".join([cl.name for cl in cl_launchers.keys()])))
+		
+		cl_launcher, launch_method = cl_launchers[cl_client_type]["launcher"], cl_launchers[cl_client_type]["launch_method"]
+		cl_service_id = "{0}{1}".format(CL_CLIENT_SERVICE_ID_PREFIX, index)
+
+		new_cl_node_validator_keystores = preregistered_validator_keys_for_nodes[index]
+
+		el_client_context = all_el_client_contexts[index]
+
+		mev_boost_context = None
+
+		if proto.has(participant, "builder_network_params"):
+			mev_boost_launcher = new_mev_boost_launcher(MEV_BOOST_SHOULD_CHECK_RELAY, participant.builder_network_params.relay_endpoints)
+			mev_boost_service_id = MEV_BOOST_SERVICE_ID_PREFIX.format(1)
+			mev_boost_context = launch_mevboost(mev_boost_launcher, mev_boost_service_id, network_params.network_id)
+
+		all_mevboost_contexts.append(mev_boost_context)
+
+		cl_client_context = None
+
+		if index == 0:
+			cl_client_context = launch_method(cl_launcher, cl_service_id, participant.cl_client_image, participant.cl_client_log_level, global_log_level, CL_CLIENT_CONTEXT_BOOTNODE,
+              el_client_context, mev_boost_context, new_cl_node_validator_keystores, participant.beacon_extra_params, participant.validator_extra_params)
```

Comment:
Not sure about the formatting here, but maybe putting one arg per line would make this a bit more readable